src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((currentGood, nextGood) => (
+        currentGood.localeCompare(nextGood)
+      ));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((currentGood, nextGood) => (
+        currentGood.length - nextGood.length
+      ));
+      break;
+
+    case SortType.NONE:
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSort] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+  const visibleGood = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+  const shouldRenderResetButton = sortType !== SortType.NONE || isReversed;
+  const resetButton = () => {
```

Comment:
The variable name looks like there is a button component, the function name should say what this function does